prometheus.lua

```diff
@@ -90,10 +152,14 @@ local function full_metric_name(name, label_names, label_values)
   end
   local label_parts = {}
   for idx, key in ipairs(label_names) do
-    local label_value = (string.format("%s", label_values[idx])
-      :gsub("[^\032-\126]", "")  -- strip non-printable characters
-      :gsub("\\", "\\\\")
-      :gsub('"', '\\"'))
+    local label_value = ""
+    if type(label_values[idx]) == "string" then
+      local valid, pos = validate_utf8_string(label_values[idx])
+      label_value = (valid and label_values[idx] or string.sub(label_values[idx], 1, pos - 1))
```

Comment:
Is this necessary for performance, or did you just choose a more compact way to express this?

I think expanding this to a more standard if/else block will make it more readable.

Reply:
Thanks, I just made things more complicated.